command/agent_test.go

```diff
@@ -514,6 +515,256 @@ listener "tcp" {
 	}
 }
 
+// TestAgent_Template_UserAgent Validates that the User-Agent sent to Vault
+// as part of Templating requests is correct.
```

Comment:
I went through this test but I am confused where we are validating User-Agent header here? Sorry, if I missed something here

Reply:
No worries, this is admittedly a bit of a weird test, but this seemed to be the best way to test it.

These tests use a custom handler func:
```
			HandlerFunc: vaulthttp.HandlerFunc(
				func(properties *vault.HandlerProperties) http.Handler {
					h.props = properties
					h.userAgentToCheckFor = useragent.AgentTemplatingString()
					h.pathToCheck = "/v1/secret/data"
					h.requestMethodToCheck = "GET"
					h.t = t
					return &h
				}),
```
Using the `userAgentHandler` in the same file.

In particular, when Vault receives a request that matches the paths/method, it can check the request's user agent.

Reply:
I'll improve the comment to point this out :)